packages/pyright-internal/src/languageService/completionProvider.ts

```diff
@@ -2669,10 +2674,20 @@ export class CompletionProvider {
                                         } else if (isOverloadedFunction(functionType)) {
                                             // 35 is completion tooltip's default width size
                                             typeDetail = getOverloadedFunctionTooltip(
+                                                /* label */ '',
                                                 functionType,
                                                 this._evaluator,
+                                                this._configOptions.compactSignatureDisplay,
                                                 /* columnThreshold */ 35
                                             );
+                                        } else if (isFunction(functionType)) {
+                                            typeDetail = getFunctionTooltip(
+                                                '',
+                                                name,
+                                                functionType,
+                                                this._evaluator,
+                                                this._configOptions.compactSignatureDisplay
+                                            );
                                         } else {
                                             typeDetail = name + this._evaluator.printType(functionType);
                                         }
```

Comment:
below this code we handle class constructor completions.....

its interesting that we dont actually try to show the true constructor and just assume no params `foo()`

```typescript
		 case DeclarationType.Class:
	         case DeclarationType.SpecialBuiltInClass: {
	             typeDetail = 'class ' + name + '()';
	             break;
           }
```